### PR TITLE
Add text_key and label_key to TextClassificationJsonReader

### LINKED ISSUE #4636: What types of sequence labeling models are possible to train?
I'm having a bit out trouble finding what types of sequence labeling models, in particular, for NER, are possible to train using AllenNLP?

In other words, what are all the architectures that one can use? For example, BI-LSTM-CRF.

### LINKED ISSUE #4936: Issue:  text_classification_json dataset class does not do what it says it does
Unfortunately, I can't find any examples of what it "does" do, but it appears to take a file that is NOT in json format, but has a complete json representation on every single line of the file, with two keys:  "text" and "label".

The code showing that is in its _read method:
....
def _read(self, file_path):
        with open(cached_path(file_path), "r") as data_file:
            for line in self.shard_iterable(data_file.readlines()):
                if not line:
                    continue
                items = json.loads(line)
                text = items["text"]
                label = items.get("label")
...
 

in other words, every non-blank single line of the file must have:
{"text": "my sample text", "label": "my_label"}
 or it fails. 

So it really should be re-labelled to reflect what it really does, i.e. every non-blank line .

So I am writing a subclass of it that does really do what it claims to do (i.e. read a classification dataset that is in json format), if anybody is interested.  The subclass will include properties for:

instance_label: str=None :  key at top level of Json file that returns a list of document instances.  If blank, then the json file is considered to be in json list format (i.e. [ {"text": "text1", "label": "label1"}, ... ]
text_label: str="text" :  Each document instance in the json file should have a key to refer to the text using this.  This will fill in the 'text' field in each output instance.
label_label: str="label": Each document instance in the json file should have a key to refer to the label using this.  This will fill in the 'label' field in each output instance.

I believe that the only methods I will have to override are _read and __init__   

Would anybody be interested in this?  I can post the code for it (after I have written it and tested it out), if you are.

### PR DESCRIPTION
**Pull request by RyujiTamaki**
Fixes #4936 .

Changes proposed in this pull request:

- Add `text_key` and `label_key` to `TextClassificationJsonReader`.
- Add `test_fixtures/data/text_classification_json/ag_news_corpus_test_key_name.jsonl` for test.
- Add test.

### CODE REVIEW

File: allennlp/data/dataset_readers/text_classification_json.py [hunk 1]
```diff
@@ -44,6 +43,10 @@ class TextClassificationJsonReader(DatasetReader):
     skip_label_indexing : `bool`, optional (default = `False`)
         Whether or not to skip label indexing. You might want to skip label indexing if your
         labels are numbers, so the dataset reader doesn't re-number them starting from 0.
+    text_key: `str`, optional (default=`text`)
+        key name for TextField instance in JSON.
+    label_key: `str`, optional (default=`label`)
+        key name for LabelField instance in JSON.
```
Comment (epwalsh) [inline allennlp/data/dataset_readers/text_classification_json.py:49]:
```suggestion
    text_key: `str`, optional (default=`"text"`)
        The key name of the source field in the JSON data file.
    label_key: `str`, optional (default=`"label"`)
        The key name of the target field in the JSON data file.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,6 +18,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added a way to specify extra parameters to the predictor in an `allennlp predict` call.
 - Added a way to initialize a `Vocabulary` from transformers models.
 - Added an example for fields of type `ListField[TextField]` to `apply_token_indexers` API docs.
+- Added `text_key` and `label_key` to `TextClassificationJsonReader` class.
```
Comment (epwalsh) [inline CHANGELOG.md:21]:
```suggestion
- Added `text_key` and `label_key` parameters to `TextClassificationJsonReader` class.
```

**Review by epwalsh — CHANGES_REQUESTED** [review]
Thanks @RyujiTamaki! This looks good. I just have a few comments:

I actually don't think the test case you added is necessary. The existing tests should already cover your changes. Could you please remove that and look at the two other suggestions I made? Then we'll get this merged.

**Comment by RyujiTamaki** [comment]
@epwalsh
Thank you for reviewing! I fixed it.
But the CI has failed. It doesn't seem to have anything to do with my fix. Could you check it?
I cannot find the way to solve that.

**Comment by epwalsh** [comment]
@RyujiTamaki ah yeup, looks like our GPU CI server was having issues. I just restarted it.

**Review by epwalsh — APPROVED** [review]
LGTM! Thanks @RyujiTamaki!

**Pull request merged**